src/FunctionApps/python/IntakePipeline/fhir.py

```diff
@@ -37,4 +42,158 @@ def read_fhir_bundles(container_url: str, container_prefix: str) -> Iterator[dic
+    """
+    if len(fhir_json) > 0:
+        pass
+    elif os.path.exists(fhir_filepath):
+        fhir_string = open(fhir_filepath).read()
+        fhir_json = json.loads(fhir_string)
+    else:
+        fhir_json = json.loads(fhir_string)
+
+    import_to_fhir(fhir_json)
+
+
+def import_to_fhir(fhir_json: dict, method: str = "PUT"):
+    """Import a FHIR resource to the FHIR server.
+    The submissions may Bundles or individual FHIR resources.
+
+    See :py:func:`_ensure_bundle_batch` for details about Bundle
+    conversion and handling.
+
+    :param dict fhir_json: FHIR resource in json format.
+    :param str method: HTTP method to use (currently PUT or POST supported)
+    """
+    try:
+        token = get_access_token()
+    except Exception:
+        logging.exception("Failed to get access token")
+        raise requests.exceptions.HTTPError(
+            "Authorization error occurred while processing information into \
+            FHIR server."
+        )
+    retry_strategy = Retry(
+        total=3,
+        status_forcelist=[429, 500, 502, 503, 504],
+        allowed_methods=["HEAD", "PUT", "POST", "OPTIONS"],
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    http = requests.Session()
+    http.mount("https://", adapter)
+    http.mount("http://", adapter)
+    fhir_url = os.environ.get("FHIR_URL", "")
+    try:
+        resource_type = fhir_json["resourceType"]
+
+        if resource_type == "Bundle":
+            transaction_json = _ensure_bundle_batch(fhir_json, method)
+            requests.post(
+                fhir_url,
+                headers={
+                    "Authorization": f"Bearer {token}",
+                    "Accept": "application/fhir+json",
+                    "Content-Type": "application/fhir+json",
+                },
+                data=json.dumps(transaction_json),
+            )
+        elif method == "POST":
+            requests.post(
+                f"{fhir_url}/{resource_type}",
+                headers={
+                    "Authorization": f"Bearer {token}",
+                    "Accept": "application/fhir+json",
+                    "Content-Type": "application/fhir+json",
+                },
+                data=json.dumps(fhir_json),
+            )
+        elif method == "PUT":
+            resource_id = fhir_json["id"]
+            requests.put(
+                f"{fhir_url}/{resource_type}/{resource_id}",
+                headers={
+                    "Authorization": f"Bearer {token}",
+                    "Accept": "application/fhir+json",
+                    "Content-Type": "application/fhir+json",
+                },
+                data=json.dumps(fhir_json),
+            )
+
+    except Exception:
+        logging.exception(
+            "Request using method " + method + " failed for json: " + str(fhir_json)
+        )
+        return
+
+
+def _ensure_bundle_batch(fhir_json: dict, method: str) -> dict:
```

Comment:
I think this may be overly defensive for where we're at right now - maybe skip for now?

Reply:
There are a few different commonly used bundle types.  I wasn't sure what they look like coming in...  I'll try to confirm what they look like - if no modifications are needed (or if those modifications can be incorporated earlier in the process), I'll remove this code.

Reply:
The bundle modification code has been removed.  We confirmed the incoming data will be of type: "batch" and have the correct request data associated (PUT).  This seems to be how the bundles are coming back from the MS FHIR converter.  I did some basic testing locally, and the info seemed to be uploaded as expected without modifications.